### PR TITLE
feat(federation): provide UI toggles in Admin settings for federation options

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Ref #11503 
* Ref #11279 

### 🖼️ Screenshots / Screencasts

| 🏚️ Before | 🏡 After |
|------------|----------|
| image          | image      |

### 🏁 Checklist

- [x] 🌏 Tested with Chrome, Firefox and Safari or should not be risky to browser differences
- [ ] 🖌️ Design was reviewed, approved or inspired by the design team
- [ ] ⛑️ Tests are included or not possible
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required

---

### 🏁 Checklist

- [ ] ⛑️ Tests (unit and/or integration) are included or not possible
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/AdminSettings/Federation.vue [hunk 1]
```diff
@@ -33,28 +33,126 @@
...
+				type="switch"
+				@update:checked="saveFederationOutgoingEnabled">
+				{{ t('spreed', 'Allow users to invite federated users into conversation') }}
+			</NcCheckboxRadioSwitch>
+
+			<NcCheckboxRadioSwitch :checked="isFederationOnlyTrustedServersEnabled"
+				:disabled="loading"
+				type="switch"
+				@update:checked="saveFederationOnlyTrustedServersEnabled">
+				{{ t('spreed', 'Allow only federation with trusted servers') }}
```
Comment (SystemKeeper) [inline src/components/AdminSettings/Federation.vue:58]:
```suggestion
				{{ t('spreed', 'Allow federation with trusted servers only') }}
```
🤔

Comment (SystemKeeper) [reply src/components/AdminSettings/Federation.vue:58]: Or maybe even 
```suggestion
				{{ t('spreed', 'Only allow federation with trusted servers') }}
```

Comment (SystemKeeper) [reply src/components/AdminSettings/Federation.vue:58]: Also in the option to enable federation it’s „Federation“ with a capital „F“. Guess we should keep it consistent either way

File: src/components/AdminSettings/Federation.vue [hunk 2]
```diff
@@ -33,28 +33,126 @@
...
+			allowedGroups: [],
+			saveLabelAllowedGroups: t('spreed', 'Save changes'),
+			debounceSearchGroup: () => {},
 		}
 	},
 
+	computed: {
+		trustedServersLink() {
+			const href = generateUrl('/settings/admin/sharing#ocFederationSettings')
+			return t('spreed', 'Trusted servers could be configured at {linkstart}Sharing settings page{linkend}.')
```
Comment (SystemKeeper) [inline src/components/AdminSettings/Federation.vue:139]:
```suggestion
			return t('spreed', 'Trusted servers can be configured at {linkstart}Sharing settings page{linkend}.')
```

File: src/components/AdminSettings/Federation.vue [hunk 3]
```diff
@@ -33,28 +33,126 @@
...
+			<div class="form">
+				<NcSelect v-model="allowedGroups"
+					input-id="allow_groups_invite_federated"
+					:input-label="t('spreed', 'Limit invite federated users')"
+					name="allow_groups_invite_federated"
+					class="form__select"
+					:options="groups"
+					:placeholder="t('spreed', 'Limit invite federated users')"
+					:disabled="loading"
+					multiple
```
Comment (DorraJaouad) [inline src/components/AdminSettings/Federation.vue:74]:
```suggestion
				<NcSelect v-model="allowedGroups"
					input-id="allow_groups_invite_federated"
					:input-label="t('spreed', 'Groups allowed to invite federated users')"
					name="allow_groups_invite_federated"
					class="form__select"
					:options="groups"
					:placeholder="t('spreed', 'Select groups...')"
					:disabled="loading"
					multiple
```

File: src/components/AdminSettings/Federation.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/AdminSettings/Federation.vue:58]: I though of `Federation` as a feature in general, and `federation` as a single conversation.

Maybe `Only allow to federate with trusted servers` would be better here?

File: src/components/AdminSettings/Federation.vue [hunk 4]
```diff
@@ -33,28 +33,126 @@
...
+			debounceSearchGroup: () => {},
 		}
 	},
 
+	computed: {
+		trustedServersLink() {
+			const href = generateUrl('/settings/admin/sharing#ocFederationSettings')
+			return t('spreed', 'Trusted servers could be configured at {linkstart}Sharing settings page{linkend}.')
+				.replace('{linkstart}', `<a target="_blank" rel="noreferrer nofollow" class="external" href="${href}">`)
+				.replaceAll('{linkend}', ' ↗</a>')
```
[the remark this answers is not in the document]
Comment (Antreesy) [reply src/components/AdminSettings/Federation.vue:141]: Library is escaping special characters. So this approach is working so far in some other places

**Review by DorraJaouad — APPROVED** [review]

**Pull request merged**